src/Runner.Worker/ActionRunner.cs

```diff
@@ -89,7 +89,15 @@ public async Task RunAsync()
                 string.Equals(localAction.RepositoryType, Pipelines.PipelineConstants.SelfAlias, StringComparison.OrdinalIgnoreCase))
             {
                 var actionManager = HostContext.GetService<IActionManager>();
-                var prepareResult = await actionManager.PrepareActionsAsync(ExecutionContext, compositeHandlerData.Steps, ExecutionContext.Id);
+                PrepareResult prepareResult = default(PrepareResult);
+                try
+                {
+                    prepareResult = await actionManager.PrepareActionsAsync(ExecutionContext, compositeHandlerData.Steps, ExecutionContext.Id);
+                }
+                catch (WebApi.FailedToResolveActionDownloadInfoException ex)
+                {
+                    throw new WebApi.FailedToResolveActionDownloadInfoException("Failed to resolve action download info", ex);
```

Comment:
why do we need to catch the exception and rethrow the same exception with a different message?